cfme/tests/automate/test_instance.py

```diff
@@ -354,3 +358,130 @@ def test_quota_source_value(request, entity, search, copy_quota_instance, generi
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script="""
+               \n$evm.log(:info, "Hello from method of parent instance")
+                \nexit MIQ_STOP
+                """
+    )
+
+    # Adding parent instance to call automate method and child instances in other class
+    instance = klass.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields={STATE3: {"value": f"METHOD::{method.name}"}}
+    )
+
+    # Creating other class for two child instances
+    klass2 = namespace.classes.create(
+        name=fauxfactory.gen_alpha(),
+        display_name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha()
+    )
+    klass2.schema.add_fields(*[{'name': state, 'type': 'State'}
+                               for state in [STATE1, STATE2, STATE3]])
+
+    yield klass2, instance
+    method.delete_if_exists()
+    instance.delete_if_exists()
+    klass2.delete_if_exists()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1441353])
+@pytest.mark.parametrize("process", ["miq_stop", "miq_abort"])
+def test_miq_stop_abort_with_state_machines(request, setup, process, domain, klass, namespace):
+    """
+    Bugzilla:
+        1441353
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+    """
+    klass2, instance = setup
+
+    # Creating three child methods to execute via child instances
+    child_method = list(map(lambda num: klass2.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script=f"""
+               \n$evm.log(:info, "This is method {num}")
+                \nexit {process.upper()}
+                """
+    ), ["first", "second", "third"]))
+
+    fields = [{STATE1: {'value': f"METHOD::{child_method[0].name}"},
+              STATE2: {'value': f"METHOD::{child_method[1].name}"}},
+              {STATE1: {'value': f"METHOD::{child_method[2].name}"}}]
+
+    # Creating two child instances
+    child_inst = list(map(lambda field: setup[0].instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields=field
+    ), fields))
+
+    @request.addfinalizer
+    def finalize():
+        for method in child_method:
+            method.delete_if_exists()
+        for inst in child_inst:
+            inst.delete_if_exists()
+
+    # Updating parent instance fields to execute child instances
+    with update(instance):
+        instance.fields = {
+            STATE1: {
```

Comment:
Thoughts on using comprehension here?

Reply:
Hey @john-dupuy 
- I tried to use comprehension here but getting error. 
```
ipdb> instace_fields ={state: { 
                "value": f"/{domain.name}/{namespace.name}/{klass2.name}/{child_inst[i].name}" 
           } for i, state in enumerate([state1, state2])} 
                                                                                                                                                                             
*** NameError: name 'domain' is not defined
ipdb> domain.name                                                                                                                                                            
'VdQbatotQz'                                                                                                                                                                        
```
- hence keeping as it is.